junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/EngineDiscoveryOrchestrator.java

```diff
@@ -72,18 +75,40 @@ public EngineDiscoveryOrchestrator(Iterable<TestEngine> testEngines,
 	 * {@linkplain TestDescriptor#prune() prunes} the resulting test tree.
 	 */
 	public LauncherDiscoveryResult discover(LauncherDiscoveryRequest request, Phase phase) {
+		Map<TestEngine, TestDescriptor> result = discover(request, phase, UniqueId::forEngine);
+		return new LauncherDiscoveryResult(result, request.getConfigurationParameters());
+	}
+
+	/**
+	 * Discovers tests for the supplied request in the supplied phase using the
+	 * configured test engines.
+	 *
+	 * <p>Applies {@linkplain org.junit.platform.launcher.EngineFilter engine
+	 * filters} and {@linkplain PostDiscoveryFilter post-discovery filters} and
+	 * {@linkplain TestDescriptor#prune() prunes} the resulting test tree.
+	 *
+	 * Additionally test engines without tests are pruned from from the discovery
+	 * result and the engines unique id will be prefixed with {@code parentId}.
+	 */
+	public LauncherDiscoveryResult discover(LauncherDiscoveryRequest request, Phase phase, UniqueId parentId) {
+		Map<TestEngine, TestDescriptor> result = discover(request, phase, parentId::appendEngine);
+		return new LauncherDiscoveryResult(pruneEngines(result), request.getConfigurationParameters());
```

Comment:
Pruning the test engines is necessary because their test descriptors no longer root containers. This results in them being pruned when added as children to the parent engine. However because the `LauncherDiscoveryResult` is executed as a whole events for these empty test engine are still emitted resulting exceptions when trying to look up these nodes in in the test plan.